virtual_rainforest/models/soil/soil_model.py

```diff
@@ -117,6 +121,8 @@ def __init__(
         """The layer in the data object representing the first soil layer."""
```

Comment:
I noticed that in the `AbioticModel`, the layer roles are assigned as an attribute (`self.layer_roles`), does that make a difference?

Reply:
I chose not to assign them as an attribute as I didn't think I would use them elsewhere, i.e. I would just make use of `self.top_soil_layer_index`. If that changes I would definitely take the approach you've used in your abiotic models